clients/ctl/admin-ui/src/features/taxonomy/helpers.ts

```diff
@@ -20,10 +20,21 @@ export const transformTaxonomyEntityToNodes = (
     const thisLevelKey = thisLevelEntity.fides_key;
     return {
       value: thisLevelEntity.fides_key,
-      label: thisLevelEntity.name ?? thisLevelEntity.fides_key,
+      label:
+        thisLevelEntity.name === "" || thisLevelEntity.name == null
+          ? thisLevelEntity.fides_key
+          : thisLevelEntity.name,
       description: thisLevelEntity.description,
       children: transformTaxonomyEntityToNodes(entities, thisLevelKey),
     };
   });
   return nodes;
 };
+
+export const parentKeyFromFidesKey = (fidesKey: string) => {
+  const split = fidesKey.split(".");
```

Comment:
I'm tempted to polyfill `string` to have fides key logic methods 😆

Reply:
oooh fancy ✨ will keep in mind if we keep building stuff off of fides key!